jasmine/spec/inverted-index-test.js

```diff
@@ -1,4 +1,138 @@
+    },
+    {
+      "title": "The Lord of the Rings: The Fellowship of the Ring.",
+      "text": "An unusual alliance of man, elf, dwarf, wizard and hobbit seek to destroy a powerful ring."
+    }
+  ]`;
+  const badJSON = `[
+    {
+      "A Good Book": "Great Expectations",
+      "Description": "Pip's great expectations"
+    },
+    {
+      "A Another Good Book": "Becoming Beyonce",
+      "Description": "The untold story as told by J. Randy Taraborrelli."
+    }
+  ]`;
+
+  const invalid = '[]';
+
+  beforeEach(() => {
+    index = new Index();
+    index.createIndex('goodJSON', goodJSON);
+    index.createIndex('invalid', invalid);
+    index.createIndex('badJSON', badJSON);
+  });
+
+  /**
+   * Index instance test suit
+   */
+  describe('Inverted Index Instance', () => {
+    describe('If instantiated', () => {
+      it('should be an instance of the Index class', () => {
+        expect(index instanceof Index).toBeTruthy();
+      });
+    });
+  });
+
+  /**
+   * Read book data test suite
+   */
+
+  describe('Read book data', () => {
+    it('should ensure that JSON file is not empty', () => {
+      expect(goodJSON.length > 0).toBeTruthy();
+    });
+    it('reads valid and populated JSON files', () => {
+      expect(index.checkJSON(goodJSON)).toBeTruthy();
+    });
+    it('Should throw an error if a json file is empty or badly formatted', () => {
+      expect(() => {
+        index.checkJSON('invalid', invalid);
+      }).toThrow(new Error('Invalid JSON file! Please ensure it is properly formatted and try again. Thank you'));
+    });
+    it('rejects an invalid file', () => {
+      expect(index.createIndex('invalid', invalid)).toBeFalsy();
+    });
+    it('asserts that JSON files have title-text pairs as object keys', () => {
+      expect(index.createIndex('badJSON', badJSON)).toBeFalsy();
+    });
+  });
+
+  /**
+   * Populate index test suite
+   */
+  describe('Populate Index', () => {
+    it('reads file when an index is created', () => {
+      expect(index.getIndex('goodJSON').length).not.toEqual(0);
+      // expect(index.getIndex('invalid').length).toBeEqual(0);
+    });
+  });
+
+  /**
+   * Search index test suite
+   */
+  describe('Search Index', () => {
+    let searchResults;
+    it('does a single-word search', () => {
+      searchResults = index.searchIndex('goodJSON', 'alice');
+      expect(Array.isArray(searchResults.goodJSON)).toBeFalsy();
+      expect(typeof searchResults.goodJSON).toEqual('object');
+      expect(searchResults.goodJSON.alice).toEqual([0]);
+    });
+    it('should ensure it does a multi-word search', () => {
+      searchResults = index.searchIndex('goodJSON', 'alice rings ');
+      expect(Array.isArray(searchResults.goodJSON)).toBeFalsy();
+      expect(typeof searchResults.goodJSON).toEqual('object');
+      expect(searchResults.goodJSON.alice && searchResults.goodJSON.rings).toEqual([0] && [1]);
+    });
+    it('it successfully searches an array of words', () => {
```

Comment:
Nitpick: You should remove the "it" starting the string so that the sentence doesn't read "it it successfully........"